drivers/entropy/entropy_cc310.c

```diff
@@ -29,12 +29,29 @@ static int entropy_cc310_rng_get_entropy(struct device *dev, uint8_t *buffer,
 	__ASSERT_NO_MSG(buffer != NULL);
 
 #if defined(CONFIG_SPM)
+	size_t offset = 0;
+	size_t to_copy;
+	uint8_t spm_buf[144];
+
 	/** This is a call from a non-secure app that enables secure services,
 	 *  in which case entropy is gathered by calling through SPM
 	 */
-	res = spm_request_random_number(buffer, length, &olen);
-	if (olen != length) {
-		return -EINVAL;
+	while (length > 0) {
+		res = spm_request_random_number(spm_buf, sizeof(spm_buf),
+						&olen);
+		if (olen != sizeof(spm_buf)) {
```

Comment:
You should probably put this `if` below the other one, since `olen` might be garbage when the function returns an error.